core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.ts

```diff
@@ -198,19 +220,15 @@ export class DotBlockEditorComponent implements OnInit, OnDestroy {
      */
     private async getCustomRemoteExtensions(): Promise<AnyExtension[]> {
         const data: RemoteCustomExtentions = this.getParsedCustomBlocks();
-
-        const extensionUrls = data.extensions.map((extension) => extension.url);
+        const extensionUrls = data?.extensions.map((extension) => extension.url);
         const customModules = await this.loadCustomBlocks(extensionUrls);
         const blockNames = [];
 
         data.extensions.forEach((extension) => {
-            blockNames.push(...extension.actions.map((item) => item.name));
+            blockNames.push(...(extension.actions?.map((item) => item.name) || []));
         });
 
-        const moduleObj = customModules.reduce(
-            (prevModule, module) => ({ ...prevModule, ...module }),
-            {}
-        );
+        const moduleObj = customModules.reduce(this.parsedCustomModules, {});
```

Comment:
We need to be more robust here... for example, if a user adds the `customBlocks` variable, and it doesn't have the right properties, this will break, example:

<img width="603" alt="image" src="https://user-images.githubusercontent.com/751424/222732640-004c9628-b688-4eb8-b31c-0b38e23fc8e6.png">

I think we can do better checking here before doing the maps and if we found something wrong, we can warn the users like:

> Should have a `experiment` property array
> `experiment` property should be an array of bla bla bla
> `actions` should be an array of bla bla bla.

We can do some extra here to give the user more information on debugging.

Reply:
Added this detailed warning:


<img width="1320" alt="image" src="https://user-images.githubusercontent.com/3438705/222807053-bd62d5cc-b99e-48af-b8ed-ab9a65dfc94f.png">

Also validate the schema using this library, example:

```
    import { assert, object, string, array, is } from "superstruct";

    /**
     * assert call throws a detailed error
     * @param data 
     * @returns if the schema is valid to use
     */
    private isValidSchema(data: RemoteCustomExtensions): void {
        const RemoteExtensionsSchema = object({
            extensions: array(
              object({
                url: string(),
                actions: array(
                  object({
                    name: string(),
                    command: string(),
                    menuLabel: string(),
                    icon: string()
                  })
                )
              })
            )
          });

          assert(data, RemoteExtensionsSchema)
    }
```

Size information for `superstruct`: https://bundlephobia.com/package/superstruct@1.0.3

Reply:
3.12 MB  <- 3.05MB